### DISCUSSION TITLE
Missing dependencies / canary versions

### DISCUSSION THREAD

**Discussion by bbung** [nextauthjs/next-auth#2587] [Help]
### Description 🐜

dependencies:

    "@next-auth/prisma-legacy-adapter": "0.0.1-canary.127",
    "@next-auth/typeorm-legacy-adapter": "0.0.2-canary.129",

in package.json from Version 3.28.0 are not published on npm. 

Same applies for previous released versions. They are dependent on not existing versions from the same package.

Thanks for looking into it. 
  

### Is this a bug in your own project?

No

### How to reproduce ☕️

npm i -S next-auth

### Screenshots / Logs 📽

  npm i -S next-auth
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @next-auth/typeorm-legacy-adapter@0.0.2-canary.129.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'next-auth'

### Environment 🖥

 System:
    OS: Linux 4.19 Ubuntu 20.04.2 LTS (Focal Fossa)
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Memory: 11.72 GB / 12.32 GB
    Container: Yes
    Shell: 5.0.17 - /usr/bin/bash
  Binaries:
    Node: 14.17.4 - ~/.nvm/versions/node/v14.17.4/bin/node
    npm: 6.14.14 - ~/.nvm/versions/node/v14.17.4/bin/npm
  npmPackages:
    next: 11.1.0 => 11.1.0
    react: 17.0.2 => 17.0.2

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment]
It is not true, see here:

https://www.npmjs.com/package/@next-auth/typeorm-legacy-adapter/v/0.0.2-canary.129

If I had to guess, you use Yarn PnP, which notoriously behaves strangely: #1034, #1306, https://github.com/vercel/next.js/issues/20955

Although we *COULD* update internally to `@next-auth/typeorm-legacy-adapter@0.1.4` and `@next-auth/prisma-legacy-adapter@0.1.2` which probably solved some issues. cc @ndom91, could you open a PR and update these against `main`?

**Reply by bbung** [reply]
Actually i am just using plain old npm :) And you are right, the package appeared again. Now the other canary version package is missing.
https://www.npmjs.com/package/@next-auth/prisma-legacy-adapter@0.0.1-canary.127.

Also using the non canary versions sounds about right.

Thanks for you quick reply.

**Reply by balazsorban44** [reply]
Your link is wrong, here it is https://www.npmjs.com/package/@next-auth/prisma-legacy-adapter/v/0.0.1-canary.127

**Reply by bbung** [reply]
Thanks for looking into it and sorry for bothering you. It seems that out internal npm repo will not forward canary versions.

**Reply by balazsorban44** [reply]
we should update them soon, I can make sure. hope that helps. 😊

**Reply by ndom91** [reply]
Opened a PR with these updates here - https://github.com/nextauthjs/next-auth/pull/2589

**Reply by balazsorban44** [reply]
@bbung we released `3.29.0`, which shouldn't use the `canary` versions anymore!

**Comment by balazsorban44** [comment]
For further investigation, we will need a reproduction
